diff --git a/dist/commonjs/native/styles/index.js b/dist/commonjs/native/styles/index.js
index 222296d6bd95a69dd87af100d8387753c54d2095..4eb4b0e61f941a0a6fa0b6f80cc34917bc665aa5 100644
--- a/dist/commonjs/native/styles/index.js
+++ b/dist/commonjs/native/styles/index.js
@@ -373,7 +373,8 @@ function nativeStyleMapping(config, props) {
       continue;
     }
     let target = props;
-    const tokens = path.split(".");
+    const resolvedPath = typeof path === 'string' ? path : key;
+    const tokens = resolvedPath.split(".");
     const lastToken = tokens.pop();
     for (const token of tokens) {
       target[token] ??= {};
diff --git a/dist/module/native/styles/index.js b/dist/module/native/styles/index.js
index 067f8edb6459cf6b15258df079c541746f233af5..50d4c854ceacf54cfd218fd88aa55e5d549691fe 100644
--- a/dist/module/native/styles/index.js
+++ b/dist/module/native/styles/index.js
@@ -369,7 +369,8 @@ function nativeStyleMapping(config, props) {
       continue;
     }
     let target = props;
-    const tokens = path.split(".");
+    const resolvedPath = typeof path === 'string' ? path : key;
+    const tokens = resolvedPath.split(".");
     const lastToken = tokens.pop();
     for (const token of tokens) {
       target[token] ??= {};
diff --git a/src/native/styles/index.ts b/src/native/styles/index.ts
index 9ac5cc4d7c532cd9f6686d4b9d5909d5a76e8aab..85f0eb691a3d177db552959fd8013900258e5e4e 100644
--- a/src/native/styles/index.ts
+++ b/src/native/styles/index.ts
@@ -504,7 +504,8 @@ function nativeStyleMapping(
     }
 
     let target = props;
-    const tokens = path.split(".");
+    const resolvedPath = typeof path === 'string' ? path : key;
+    const tokens = resolvedPath.split(".");
     const lastToken = tokens.pop();
     for (const token of tokens) {
       target[token] ??= {};
